### PR TITLE
Check for vt-x support before downloading zentyal vm image 

### PR DESCRIPTION
**Pull request by 650elx**

### CODE REVIEW

File: src/dispatcher/plugins/DomainControllerPlugin.py [hunk 1]
```diff
@@ -124,7 +124,13 @@ def describe(self, dc):
         return TaskDescription('Configuring Domain Controller vm service')
 
     def verify(self, dc):
-        return ['system']
+        hw_capabilities = self.dispatcher.call_sync('vm.provide_hw_vm_capabilities')
+        if (hw_capabilities['vtx_enabled'] or hw_capabilities['svm_features']) and hw_capabilities['unrestricted_guest']:
```
Comment (shangteus) [inline src/dispatcher/plugins/DomainControllerPlugin.py]: Zentyal does need GRUB, so this should be: (vtx_enabled and unrestricted_guest) or svm_features

Intel with unrestricted guest OR AMD

File: src/dispatcher/plugins/VMPlugin.py [hunk 2]
```diff
@@ -233,6 +234,22 @@ def request_serial_console(self, id):
     def request_webvnc_console(self, id):
         return self.dispatcher.call_sync('containerd.console.request_webvnc_console', id)
 
+    @returns(h.object())
```
Comment (shangteus) [inline src/dispatcher/plugins/VMPlugin.py]: Maybe we could have a schema for that? These are 3 well defined fields

File: src/dispatcher/plugins/VMPlugin.py [hunk 3]
```diff
@@ -233,6 +234,22 @@ def request_serial_console(self, id):
...
+    def provide_hw_vm_capabilities(self):
```
Comment (shangteus) [inline src/dispatcher/plugins/VMPlugin.py]: That's mostly OCD, but 'get_hw_capabilities' would be fine and save the rainforest :D

File: src/dispatcher/plugins/VMPlugin.py [hunk 4]
```diff
@@ -233,6 +234,22 @@ def request_serial_console(self, id):
...
+
+            if sysctl.sysctlbyname('hw.vmm.svm.features') > 0:
+                hw_capabilities['svm_features'] = True
+
+            if sysctl.sysctlbyname('hw.vmm.vmx.cap.unrestricted_guest'):
+                hw_capabilities['unrestricted_guest'] = True
+        except OSError:
+            pass
+        return hw_capabilities
+
```
Comment (shangteus) [inline src/dispatcher/plugins/VMPlugin.py]: If we're already adding an RPC for that, then let's replace code in containerd doing the same thing with call to that RPC. Doubled code makes pandas die.

**Comment by 650elx** [comment]
Issues fixed - please review

**Pull request merged**